source/extensions/filters/http/oauth2/filter.cc

```diff
@@ -308,7 +308,7 @@ Http::FilterHeadersStatus OAuth2Filter::decodeHeaders(Http::RequestHeaderMap& he
   Http::Utility::Url state_url;
   if (!state_url.initialize(state_, false)) {
     sendUnauthorizedResponse();
-    return Http::FilterHeadersStatus::StopAllIterationAndBuffer;
```

Comment:
So I get switching the returns where we're sending a local reply, but it looks like you're switching from "stop all" to "stop" in other cases as well which I'm less confident of.

I thought oath was supposed to stop all progress until oath responded?  I think we need to revert those parts of the change.
FWIW I'd also be fine allowing any of the Stop calls if you think that's cleaner than fixing up specific returns of oath.

Reply:
Oh i missed this one:

```
  // pause while we await the next step from the OAuth server
-  return Http::FilterHeadersStatus::StopAllIterationAndBuffer;
 + return Http::FilterHeadersStatus::StopIteration;
```

yeah I agree that one is fine to buffer I guess... but probably not a biggie if we fully pause until we hear back.

Reply:
Sounds good, I'm not sure which makes sense -- I'll make the change less restrictive.